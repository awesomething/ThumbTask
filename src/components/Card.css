.card {
    margin: 10px;
    border-radius: 10px;
    overflow: hidden;
    box-shadow: 0px 6px 18px -9px rgba(0, 0, 0, 0.75);
    transition: transform 100ms ease-in;
    cursor: pointer;
    width: 25%;
    flex-direction: row;
    display: inline-block;
    justify-content: center;
    flex-wrap: wrap;
    flex-grow: 1;
    color: black;
  }
  
  .card:hover {
    transform: scale(1.07);
  }
  
  .card > img {
    object-fit: fill;
    min-width: 250px;
    min-height: 300px;
    width: 30%;
    height: 30%;
    flex-direction: column;
    display: inline;
    justify-content: center;
    flex-wrap: wrap;
    flex-grow: 1;
  }
  
  .card__info {
      margin-top: -9px;
      border-radius: 10px;
      padding: 20px;
      padding-top: 20px;
      border: 1;
      
    }
  
    .card__info > h2 {
      font-size: 18px;
      font-weight: 600;
    }
    
    .card__info > h4 {
      font-size: 14px;
      font-weight: 300;
      margin-top: 8px;
      margin-bottom: 8px;
    }
  
      
  @media (max-width: 800px) {
    .card, .card > img {
      max-width: 375px;
    }
  }