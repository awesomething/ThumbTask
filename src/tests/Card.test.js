import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Card from '../components/Card'
import {configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe(`Card component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    
  }

  it('renders a Card by default', () => {
    const wrapper = shallow(<Card />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Card given props', () => {
    const wrapper = shallow(<Card {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
