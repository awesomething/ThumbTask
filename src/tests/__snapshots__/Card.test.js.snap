// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders a Card by default 1`] = `
<div
  className="card"
>
  <img
    alt=""
  />
  <div
    className="card__info"
  >
    <h2 />
  </div>
</div>
`;

exports[`Card component renders the Card given props 1`] = `
<div
  className="card"
>
  <img
    alt=""
  />
  <div
    className="card__info"
  >
    <h2 />
  </div>
</div>
`;
